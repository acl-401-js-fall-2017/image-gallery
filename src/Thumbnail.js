import React, { PureComponent } from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types'

export default class Thumbnail extends PureComponent {
  render() {
    const {gallery, shouldDisplay, deleteImage } =this.props
    return( 
      <StyledDiv shouldDisplay={shouldDisplay}>
        <ul style ={{ display: 'flex' }}>
          {gallery.map((img, i) => (
            <div key={img._id}>
              <div className ="thumbnailDiv">
                <div className="deleteDiv" onClick ={ () => deleteImage(img._id) }>x</div>
                <img className ='galleryImg' src={img.img} alt=''/>
              </div>
            </div>
          ))}
        </ul>
      </StyledDiv>
    )
  };
}

const StyledDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
`;

Thumbnail.propTypes = {
  gallery: propTypes.array,
  shouldDisplay: propTypes.bool,
  deleteImage: propTypes.func
}