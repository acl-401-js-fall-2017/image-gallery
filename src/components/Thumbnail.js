import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Thumbnail extends PureComponent {
    
    static propTypes = {
      bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }


    render() {
      return (
        <div className="thumbnail">
          {this.props.bunnies.map((bunny, index) => {
            return (
              <div key={index}>
                <img src={bunny.url} alt={bunny.title} height="100px"/>
                <h4>{bunny.title}</h4>
                <h6>{bunny.description}</h6>
              </div>
            );
          })}
        </div>
      );
    }
}