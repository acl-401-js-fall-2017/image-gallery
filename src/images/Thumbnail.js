import React, { PureComponent } from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import { loadGallery, deleteImg } from '../utils/actions';
import imgApi from '../utils/services/imgApi';

export default class Thumbnail extends PureComponent {
  state ={
    gallery: []
  }

  getAlbumId() {
    console.log('this is id:',this.props.match.params.id);
    return this.props.match.params.id;
  }

  async componentDidMount() {
    const img = await imgApi.get(this.getAlbumId());
    console.log('img is:',img);
    const newState = loadGallery(this.state, img);
    this.setState(newState);
    console.log('thumbnail just maunted and we got',this.state);
  }

  handleDelete = async id => {
    await imgApi.remove(id);
    const newState = deleteImg(this.state, id);
    this.setState(newState);
  }


  render() {    
    return( 
      <StyledDiv>
        <ul style ={{ display: 'flex' }}>
          {this.state.gallery.map((img, i) => (
            <div key={img._id}>
              <div className ="thumbnailDiv">
                <DeleteDiv onClick={() => this.handleDelete(img._id)}>X</DeleteDiv>
                <img className ='galleryImg' src={img.url} alt=''/>
              </div>
            </div>
          ))}
        </ul>
      </StyledDiv>
    );
  }
}

const StyledDiv = styled.div`
display: 'flex';
`;

const DeleteDiv = styled.div`
margin-left: 90%;
text-align: center;
border: 1px solid black;
`;

Thumbnail.propTypes = {
  gallery: propTypes.array,
  shouldDisplay: propTypes.bool,
  deleteImage: propTypes.func
};