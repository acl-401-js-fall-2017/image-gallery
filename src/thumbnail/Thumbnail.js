import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';

export default class Thumbnail extends PureComponent {
  render () {
    const { images } = this.props;
    return (
      <div className="thumbnail">
        {images.map((image, i) => {
          return (
            <div key={i}>
              <h3>{image.title}</h3>
              <img src={image.url} alt={image.title} height="100px"/>
            </div>
          );
        })}
      </div>
    );
  } 
}

Thumbnail.propTypes = {
  images: PropTypes.array
};