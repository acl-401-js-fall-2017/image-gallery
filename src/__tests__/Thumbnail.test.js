import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Bunnies from '../images/bunnies';
import Thumbnail from '../components/Thumbnail';

describe('List', () => {
  it('lists of bunnies in thumbnail format', () => {
    const wrapper = shallow(<Thumbnail images={Bunnies}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});