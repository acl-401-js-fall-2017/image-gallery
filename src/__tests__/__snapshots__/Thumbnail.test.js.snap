// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List lists of bunnies in thumbnail format 1`] = `
<div
  className="thumbnail"
>
  <div
    key="0"
  >
    <img
      alt="Best Bunny"
      height="100px"
      src="https://s3.amazonaws.com/f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    />
    <h4>
      Best Bunny
    </h4>
    <h6>
      Just the cutest color on a bunny ever
    </h6>
  </div>
  <div
    key="1"
  >
    <img
      alt="Kind of cute Bunny"
      height="100px"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
    />
    <h4>
      Kind of cute Bunny
    </h4>
    <h6>
      Too many carbs :(
    </h6>
  </div>
  <div
    key="2"
  >
    <img
      alt="Tiny Bunny"
      height="100px"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
    />
    <h4>
      Tiny Bunny
    </h4>
    <h6>
      Cant see me
    </h6>
  </div>
</div>
`;
